kubeconfig.go

```
@@ -55,18 +56,14 @@ func New(config Config) (*KubeConfig, error) {
 // the current cluster is returned.
 func (k KubeConfig) NewG8sClientForApp(ctx context.Context, app v1alpha1.App) (versioned.Interface, error) {
 	secretName := secretName(app)
+	secretNamespace := secretNamespace(app)
 
 	// KubeConfig is not configured so connect to current cluster.
 	if secretName == "" {
 		return k.g8sClient, nil
 	}
 
-	kubeConfig, err := k.getKubeConfigFromSecret(ctx, secretName, secretNamespace(app))
-	if err != nil {
-		return nil, err
-	}
-
-	restConfig, err := clientcmd.RESTConfigFromKubeConfig(kubeConfig)
+	restConfig, err := k.NewRESTConfigForApp(ctx, secretName, secretNamespace)
 	if err != nil {
 		return nil, err
```

Please mask all errors. See https://github.com/giantswarm/fmt/blob/master/go/errors.md.